### DISCUSSION TITLE
Trying to create a map for my macbook, error on the F4 key

### DISCUSSION THREAD

**Discussion by Vermoot** [manna-harbour/xmk#8] [Q&A]
Hey there!
I'm trying to set up xmk native on my macbook so I can use a ZMK keymap on it, which will be awesome.
Trying to create a map for it, this is what I'm doing:
```
xmk on  main [?] took 26s
❯ ./src/xmk -k /dev/input/by-id/usb-Apple_Inc._Apple_Internal_Keyboard___Trackpad-event-kbd -c ./src/maps/macbook/macbook
```
Then I hit every key from top-lef to bottom-right, but doing so I get an error on the F4 key (actually `XF86LaunchB` according to `xev`).
Here is what the output looks like:
```
xmk on  main [?] took 26s
❯ ./src/xmk -k /dev/input/by-id/usb-Apple_Inc._Apple_Internal_Keyboard___Trackpad-event-kbd -c ./src/maps/macbook/macbook
esc brightnessdown brightnessup scale Traceback (most recent call last):
  File "/home/vermoot/Projects/xmk/./src/xmk", line 82, in <module>
    code = evdev.ecodes.KEY[event.code][4:].lower()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'list' object has no attribute 'lower'
```

Is that an xmk bug, or a mistake on my part?

**Comment by manna-harbour** [comment]
It looks like there can be multiple keycodes per key in some circumstances. The code will need to be updated to handle that.

As a workaround, you might be able to remove the extra keycodes from that key with `xmodmap` or similar.
